internal/gribi/gribi.go

```diff
@@ -225,11 +225,15 @@ func (c *Client) AddNH(t testing.TB, nhIndex uint64, address, instance string, e
 		}
 	case "MACwithInterface":
 		for _, opt := range opts {
-			if opt.SubInterface != 0 {
-				nh = nh.WithSubinterfaceRef(opt.Interface, opt.SubInterface).
-					WithMacAddress(opt.Mac)
-			} else {
+			if opt.Dest != "" {
 				nh = nh.WithIPAddress(opt.Dest).WithMacAddress(opt.Mac)
```

Comment:
This statement will not use interface-ref at all though.
Is this the intention?

If you want just to supplement interface-ref with an ip. addr, I think Ankur resolves this in a more appropriate way in #1762

Reply:
Correct, I was checking if we have Destination ip provided in which case we use mac + ip
else we further check if subinterface is there and use subinterface + Mac else Interface+Mac

Reply:
In that case (if interface-ref cannot be supported / Ankur's PR is not going to solve it), I'd propose to do this as a separate `case` statement in the helper.

Overloading `MACwithInterface` case with a scenario where, in fact, interface-reference is not used or present is very confusing/misleading.

Reply:
Okay, let me add another case

Reply:
thanks 👍